nuxeo-distribution/nuxeo-nxr-server/src/main/resources/templates/common-base/nxserver/config/kafka-config.xml.nxftl

```diff
@@ -8,20 +8,22 @@
       <producer>
         <property name="bootstrap.servers">${kafka.bootstrap.servers}</property>
         <property name="default.replication.factor">${kafka.default.replication.factor}</property>
-<#if "${kafka.ssl}" == "true">
-<#if "${kafka.sasl.enabled}" == "true">
+<#if ((kafka.sasl.enabled)!"false") == "true" || ((kafka.ssl)!"false") == "true">
         <property name="security.protocol">${kafka.security.protocol}</property>
+</#if>
+<#if "${kafka.sasl.enabled}" == "true">
         <property name="sasl.mechanism">${kafka.sasl.mechanism}</property>
         <property name="sasl.jaas.config">${kafka.sasl.jaas.config}</property>
-<#else>
-        <property name="security.protocol">SSL</property>
 </#if>
+<#if "${kafka.ssl}" == "true">
         <property name="ssl.truststore.type">${kafka.truststore.type}</property>
         <property name="ssl.truststore.location">${kafka.truststore.path}</property>
         <property name="ssl.truststore.password">${kafka.truststore.password}</property>
+<#if ((kafka.security.protocol)!"") == "SSL">
```

Comment:
This is not clear does this mean that:
- SASL_SSL prevents to use an SSL keystore?
- SSL always requires a keystore?

Reply:
In the case of both your bullets:
Indeed, since Kafka can work with a variety of configurations that are all independant from each other, the absence of one config shouldn't imply the presence of another.

[useful link](https://kafka.apache.org/documentation/#security_configclients)

Rather than relying on `kafka.security.protocol` to infer the presence of a keystore we should rely on a kafka.authentifcation.protocol` if anything other than SSL -> no keystore.

WDYT ?

Reply:
For me, if the `kafka.keystore.path` is empty you don't put any `ssl.keystore*` property.

Reply:
then I'll do the check on this